### ISSUE TITLE
Add support for API tokens with azure-ad enabled

### ISSUE THREAD

**Issue by ErikBergfurInfor** [jenkinsci/azure-ad-plugin#84]
We want to allow remote access to a view in our build-system but keep the rest of Jenkins locked down. Previously we were able to do this by creating a new user and then an API token for this user.
By configuring the security to only allow the user to read this specific view, we could use the access token to authenticate with Jenkins and show the view in our office build-monitor screens.

Once the azure-ad plugin is enabled, some of the access is triggering the Azure-ad login redirect, despite having provided a API token.

Is it possible to allow for authenticating with API tokens when the azure-ad plugin is enabled? If not, is there another way for us to get access to a view in a secure, non-azure-ad fashion and still require azure-ad authentication for the rest of Jenkins?

The view I want to use, is one created with the [build-monitor-plugin](https://plugins.jenkins.io/build-monitor-plugin/)
Thanks
   Erik

**Comment by walliski** [comment]
I have seen issues using the API token also. Found a workaround based on this issue: https://issues.jenkins.io/browse/JENKINS-54115

Basically in the matrix permissions, you have to add the user rights twice. One user will be the one that is found from Azure AD, which will have the name `UserName (some-random-guid-here)`. The other user will simply be the email that the user has, so it will just look like `UserName@hurrdidurr.com`. Give both users the same permissions, and it should work.

It would be nice if this issue would be fixed properly though, as this is not really intuitive at all.

**Comment by ErikBergfurInfor** [comment]
The workaround works somewhat. Some functions (Like accessing the jenkins API) seems to work. Accessing the view does not work though.

**Comment by timja** [comment]
@ErikBergfurInfor you could try https://github.com/jenkinsci/azure-ad-plugin/pull/112

**Comment by ErikBergfurInfor** [comment]
I just tried setting everything up with a simple nginx-proxy that just added an authorization-header (with the API-token) before passing the requests to the real jenkins-instance. With an old IE, I got a 404 for a Staple-request. But with a proper browser like Chrome and Firefox, it all seems to work!

This was the initial setup I had tested before filing this jira so I'm quite sure this was broken previously. But now, as I said, the work-around actually works! I'll close this issue as fixed (Don't know when) and I'll create a new one, if I encounter any more issues regarding Azure AD.
Thanks
